universal-application-tool-0.0.1/test/services/applicant/ApplicantServiceImplTest.java

```diff
@@ -161,6 +161,40 @@ public void stageAndUpdateIfValid_rawUpdatesContainMultiSelectAnswers_isOk() {
         .hasValue(ImmutableList.of(1L, 2L));
   }
 
+  @Test
+  public void stageAndUpdateIfValid_withEnumeratorChangesAndDeletes_isOk() {
+    QuestionDefinition enumeratorQuestionDefinition =
+        testQuestionBank.applicantHouseholdMembers().getQuestionDefinition();
+    createProgram(enumeratorQuestionDefinition);
+
+    Applicant applicant = subject.createApplicant(1L).toCompletableFuture().join();
+
+    Path enumeratorPath =
+        ApplicantData.APPLICANT_PATH.join(enumeratorQuestionDefinition.getQuestionPathSegment());
+    Path deletionPath = Path.empty().join(Scalar.DELETE_ENTITY).asArrayElement();
+    ImmutableMap<String, String> rawUpdates =
+        ImmutableMap.of(
+            enumeratorPath.atIndex(0).toString(), "first",
+            enumeratorPath.atIndex(1).toString(), "second",
+            enumeratorPath.atIndex(2).toString(), "third",
+            deletionPath.atIndex(0).toString(), "2",
+            deletionPath.atIndex(1).toString(), "0");
+
+    ErrorAnd<ReadOnlyApplicantProgramService, Exception> errorAnd =
+        subject
+            .stageAndUpdateIfValid(applicant.id, programDefinition.id(), "1", rawUpdates)
+            .toCompletableFuture()
+            .join();
+
+    assertThat(errorAnd.isError()).isFalse();
+    assertThat(errorAnd.getResult()).isInstanceOf(ReadOnlyApplicantProgramService.class);
+
+    ApplicantData applicantDataAfter =
+        userRepository.lookupApplicantSync(applicant.id).get().getApplicantData();
+
+    assertThat(applicantDataAfter.readRepeatedEntities(enumeratorPath)).containsExactly("second");
```

Comment:
question: I think I'm just misunderstanding, but why do we only expect "second"? I thought 1 and 0 are deleted, which just leaves "third"

Reply:
The delete array will always have indices 0...n where n is the number of things the applicant wants to delete, and we use the values of the array to determine what index to actually delete.

So `delete = [2, 0]` (on line 180 and 181), which means we delete whatever is at index 0 and index 2, which leaves "second" at index one.